src/repository/middleware.ts

```diff
@@ -0,0 +1,76 @@
+import { NextFunction, Request, Response } from 'express';
+import { v4 as uuid } from 'uuid';
+import { decrypt } from '../crypt';
+import logger from '../logger';
+import * as userService from './services/user-service';
+interface IUser {
+  name: string;
+  pass: string;
+  role: string;
+  favMovies: string[];
+}
+
+declare global {
```

Comment:
It's good that you find the way to handle the problem with missed properties in `Request` type, but using `any` will be good enough. For this course goals, this kind of simplification is acceptable.

Reply:
Yeah, I also realised that, after spending much time to investigate that, but it was to late to stop and stay with simplified